ops/model.py

```diff
@@ -423,9 +428,12 @@ def status(self, value: 'StatusBase'):
         if not self._backend.is_leader():
             raise RuntimeError('cannot set application status as a non-leader unit')
 
-        for _key in {'name', 'message'}:
-            assert isinstance(getattr(value, _key), str), f'status.{_key} must be a string'
-        self._backend.status_set(value.name, value.message, is_app=True)
+        self._backend.status_set(
+            typing.cast(_SettableStatusName, value.name),  # TODO: narrower type for value
```

Comment:
My thinking was that we could eventually narrow the type of value to help with static checking, as you've suggested, or with an intermediate base class for settable `Status` types that takes care of the type of `status.name`.

The major blocker for this currently is that the getter needs to return `StatusBase`, since it should be valid to learn that Juju has set your status to `error` or `unknown` via this mechanism, right?

Python type checkers don't seem to fully support having [different setter and getter signatures for properties](https://github.com/python/mypy/issues/3004), so even if we use a narrower type for the setter, users won't see errors when they write e.g. `my_unit.status = UnknownStatus()` (though this *will* explode at runtime).

In the meantime, I'll just leave the `cast` and remove the `TODO`.